script.js

```diff
@@ -1,4 +1,294 @@
+    Click enter to start another new game`;
+    return myOutputValue;
+  }
+};
+
+//Beat That Game Function
+var beatThatGame = function () {};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  if (currentGameMode == "welcomeToBeatThat") {
+    //Opening message + Instructions
+    myOutputValue = `Welcome to BEAT THAT!<br><br>
+     BEFORE starting the game, please ENTER desired player 1 In Game Name (IGN) and click ENTER, followed by player 2. <br><br>
+      Game Rules: <br>
+      1. This is a 2 person, player vs player game. <br>
+2. The game will roll 2 dice and show the dice rolls result. For example, dice 1 rolled 3 and dice 2 rolled 4. <br>
+3. The player that rolled the dice will pick the order of the dice.(If they wanted the number 43, they would specify that the 2nd dice goes first.) <br>
+4. After both players have rolled and chosen their dice order, the player with the higher combined number wins.  `;
+
+    currentGameMode = "enterPlayer1";
+
+    return myOutputValue;
+  }
+  if (currentGameMode == "enterPlayer1") {
+    //Set player 1 IGN
+    player1 = input;
+
+    currentGameMode = "enterPlayer2";
+
+    console.log(`currentGameMode, ${currentGameMode}`);
+    console.log(`Player 1 IGN, player 1: ${player1}`);
+
+    myOutputValue = `Welcome ${player1}. Player 2, please enter your desired IGN and click enter. `;
+  } else if (currentGameMode == "enterPlayer2") {
+    //Set player 2 IGN
+    player2 = input;
+
+    currentGameMode = "player1Roll"; //Switch to Game Result for player 1
+
+    console.log(`currentGameMode, ${currentGameMode}`);
+    console.log(`Player 2 IGN, player 2: ${player2}`);
+
+    myOutputValue = `Welcome ${player2}. <br><br>
+    ${player1}, you may begin to start the game by clicking enter.`;
+  } else if (
+    // Condition for player 1 turn
+    //player 1 turn + player 1 result
+    currentGameMode == "player1Roll" || //Dice results of player 1
+    currentGameMode == "player1Choice" //player 1 desired outcome
+  ) {
+    userChosenDiceOrder = input;
+    myOutputValue = player1Result(player1, player2, userChosenDiceOrder);
+    return myOutputValue;
+  } else if (
+    //player 2 turn + player 2 result
+    currentGameMode == "player2Roll" || //Dice results of player 2
+    currentGameMode == "player2Choice" //player 2 desired outcome
+  ) {
+    userChosenDiceOrder = input;
+    myOutputValue = player2Result(player2, userChosenDiceOrder);
+    return myOutputValue;
+  } else if (currentGameMode == "gameOutcome") {
+    myOutputValue = gameOutcome(player1, player2); // results outcome
+
+    //Reset Game Mode
+    currentGameMode = "welcomeToBeatThat";
+
+    //Reset both player dice array
+    player1dice1Arr = [];
+    player1dice2Arr = [];
+    player2dice1Arr = [];
+    player2dice2Arr = [];
+
+    //Reset player scores
+    player1Score = ""; //Player 1 score at the end of turn
+    player2Score = ""; //Player 2 score at the end of turn
```

Comment:
a separate helper function to reset your game would prob be ideal, esp if its a more complex game with more variables

for your desired specific feedback, prob not necessary to reset the player names if you are absolutely certain in your gameplay logic that it will get reset to new values when the game restarts. but otherwise, the guideline is better safe than sorry